From f2ba296053d64df74a8414816c4ef82f79f92034 Mon Sep 17 00:00:00 2001
Message-Id: <f2ba296053d64df74a8414816c4ef82f79f92034@dist-git>
From: Farhan Ali <alifm@linux.ibm.com>
Date: Mon, 3 Sep 2018 09:58:50 +0200
Subject: [PATCH] qemu: mdev: Use vfio-pci 'display' property only with
 vfio-pci mdevs

S390 is aware of both vfio-pci and vfio-ccw devices, so
on S390 the capability QEMU_CAPS_VFIO_PCI_DISPLAY will be
available. Add an extra check to make sure we only set the
display to off for vfio-pci mediated devices. Otherwise we
add display for vfio-ccw device and this breaks vfio-ccw
device qemu command line.

Fixes: d54e45b6e conf: Introduce new <hostdev> attribute 'display'
Signed-off-by: Farhan Ali <alifm@linux.ibm.com>
Reviewed-by: Marc Hartmayer <mhartmay@linux.ibm.com>
Reviewed-by: Erik Skultety <eskultet@redhat.com>
(cherry picked from commit d6f97d1338ba9470f7c745fab317d272cde84d38)

RHEL-7.6: https://bugzilla.redhat.com/show_bug.cgi?id=1624735
RHEL-8.0: https://bugzilla.redhat.com/show_bug.cgi?id=1624740
Signed-off-by: Erik Skultety <eskultet@redhat.com>
Reviewed-by: Jiri Denemark <jdenemar@redhat.com>
---
 src/qemu/qemu_domain.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/qemu/qemu_domain.c b/src/qemu/qemu_domain.c
index 508846116b..17be6e5537 100644
--- a/src/qemu/qemu_domain.c
+++ b/src/qemu/qemu_domain.c
@@ -6272,6 +6272,7 @@ qemuDomainHostdevDefMdevPostParse(virDomainHostdevSubsysMediatedDevPtr mdevsrc,
     /* QEMU 2.12 added support for vfio-pci display type, we default to
      * 'display=off' to stay safe from future changes */
     if (virQEMUCapsGet(qemuCaps, QEMU_CAPS_VFIO_PCI_DISPLAY) &&
+        mdevsrc->model == VIR_MDEV_MODEL_TYPE_VFIO_PCI &&
         mdevsrc->display == VIR_TRISTATE_SWITCH_ABSENT)
         mdevsrc->display = VIR_TRISTATE_SWITCH_OFF;
 
-- 
2.18.0

